Source/JavaScriptCore/wasm/WasmIPIntTierUpCounter.h

```diff
@@ -85,9 +86,15 @@ class IPIntTierUpCounter : public BaselineExecutionCounter {
         return entry->value;
     }
 
+    ALWAYS_INLINE CompilationStatus compilationStatus(MemoryMode mode) { return m_compilationStatus[static_cast<MemoryModeType>(mode)]; }
+    ALWAYS_INLINE void setCompilationStatus(MemoryMode mode, CompilationStatus status) { m_compilationStatus[static_cast<MemoryModeType>(mode)] = status; }
+
+    ALWAYS_INLINE CompilationStatus loopCompilationStatus(MemoryMode mode) { return m_loopCompilationStatus[static_cast<MemoryModeType>(mode)]; }
+    ALWAYS_INLINE void setLoopCompilationStatus(MemoryMode mode, CompilationStatus status) { m_loopCompilationStatus[static_cast<MemoryModeType>(mode)] = status; }
+
     Lock m_lock;
-    CompilationStatus m_compilationStatus { CompilationStatus::NotCompiled };
-    CompilationStatus m_loopCompilationStatus { CompilationStatus::NotCompiled };
+    CompilationStatus m_compilationStatus[numberOfMemoryModes];
```

Comment:
Need initialization of them since `CompilationStatus` just an enum.

Reply:
Changed.